pkg/session/content_processor.go

```diff
@@ -0,0 +1,102 @@
+package session
+
+import (
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	sep          = ","
+	timestampLen = 10
+)
+
+// ContentProcessor knows how to work with actual data in the blocklist.
+type ContentProcessor interface {
+	// Block adds the token to the blacklist, cleans up the outdated data from the secret.
+	// The method returns the updated secret and a bool that indicates if the attempt was successful.
+	Block(l *zap.SugaredLogger, secret *corev1.Secret, shortenedToken string) (*corev1.Secret, bool)
+	// IsBlocked returns true if the given shortened token is in the blocklist.
+	IsBlocked(shortenedToken string) bool
+	// UpdateCache updates the cached secret which is used for reading.
+	UpdateCache(secret *corev1.Secret)
+}
+
+type contentProcessor struct {
+	mutex        sync.Mutex
+	cachedSecret *corev1.Secret
+}
+
+func newContentProcessor(secret *corev1.Secret) ContentProcessor {
+	return &contentProcessor{
+		cachedSecret: secret,
+	}
+}
+
+// Block adds the token to the blacklist, cleans up the outdated data from the secret.
+// The method returns the updated secret and a bool that indicates if the attempt was successful.
+func (d *contentProcessor) Block(l *zap.SugaredLogger, secret *corev1.Secret, shortenedToken string) (*corev1.Secret, bool) {
+	if d.mutex.TryLock() {
```

Comment:
Why not just use `d.mutex.Lock()` and let the caller block?

Reply:
Resolving the thread, mutex is not used anymore, see https://github.com/percona/everest/pull/1293#discussion_r2058204411